jobs/furnishings/src/furnishings/sftp.py

```diff
@@ -0,0 +1,111 @@
+# Copyright © 2024 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module is a wrapper for SFTP Connection object."""
+from __future__ import annotations
+
+import io
+from enum import auto
+from typing import Callable
+
+import paramiko
+
+from furnishings.utils.base import BaseEnum
+
+
+class PublicKeyAlgorithms(BaseEnum):
+    """Cipher types."""
+
+    RSA = auto()
+    ED25519 = auto()
+
+
+class SftpConnection:
+    """SFTP Connection object."""
```

Comment:
I made some tweaks to the PPR `SftpConnection` object to make it a context manager, so it can be used as:

```
connection = SftpConnection(kwargs)
with connection as client:
    client.put(...)
```

This helps make sure the socket is closed after use. 

I also removed the previous `put` functions since we can just use the [`paramiko.SFTPClient` API](https://docs.paramiko.org/en/latest/api/sftp.html) directly.